homeassistant/components/tahoma/binary_sensor.py

```diff
@@ -0,0 +1,97 @@
+"""Support for Somfy TaHoma binary sensors."""
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_MOTION,
+    DEVICE_CLASS_OCCUPANCY,
+    DEVICE_CLASS_OPENING,
+    DEVICE_CLASS_SMOKE,
+    DOMAIN as BINARY_SENSOR,
+    BinarySensorEntity,
+)
+
+from .const import DOMAIN
+from .tahoma_entity import TahomaEntity
+
+CORE_ASSEMBLY_STATE = "core:AssemblyState"
+CORE_BUTTON_STATE = "core:ButtonState"
+CORE_CONTACT_STATE = "core:ContactState"
+CORE_GAS_DETECTION_STATE = "core:GasDetectionState"
+CORE_OCCUPANCY_STATE = "core:OccupancyState"
+CORE_OPENING_STATE = "core:OpeningState"
+CORE_OPEN_CLOSED_TILT_STATE = "core:OpenClosedTiltState"
+CORE_RAIN_STATE = "core:RainState"
+CORE_SMOKE_STATE = "core:SmokeState"
+CORE_THREE_WAY_HANDLE_DIRECTION_STATE = "core:ThreeWayHandleDirectionState"
+CORE_VIBRATION_STATE = "core:VibrationState"
+CORE_WATER_DETECTION_STATE = "core:WaterDetectionState"
+
+DEVICE_CLASS_BUTTON = "button"
+DEVICE_CLASS_GAS = "gas"
+DEVICE_CLASS_RAIN = "rain"
+DEVICE_CLASS_WATER = "water"
+
```

Comment:
Only standard entity device classes may be used. Import and use the constants from the binary sensor integration.

https://developers.home-assistant.io/docs/core/entity/binary-sensor#available-device-classes

```suggestion
```

Reply:
Will do. I should investigate if other devices could use `DEVICE_CLASS_RAIN` or `DEVICE_CLASS_WATER` as well, for a new architecture proposal.

Reply:
Fixed.